src/App/Controls/PasswordStrengthProgressBar/PasswordStrengthViewModel.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Bit.Core.Abstractions;
+using Bit.Core.Utilities;
+using Xamarin.Forms;
+
+namespace Bit.App.Controls
+{
+    public class PasswordStrengthViewModel : ExtendedViewModel
+    {
+        private readonly IPasswordGenerationService _passwordGenerationService;
+        private double _passwordStrength;
+        private Color _passwordColor;
+        private string _passwordStatus;
+
+        public PasswordStrengthViewModel()
+        {
+            _passwordGenerationService = ServiceContainer.Resolve<IPasswordGenerationService>();
+        }
+
+        public double PasswordStrength
+        {
+            get => _passwordStrength;
+            set => SetProperty(ref _passwordStrength, value);
+        }
+
+        public Color PasswordColor
+        {
+            get => _passwordColor;
+            set => SetProperty(ref _passwordColor, value);
+        }
+
+        public string PasswordStatus
+        {
+            get => _passwordStatus;
+            set => SetProperty(ref _passwordStatus, value);
+        }
+
+        public PasswordStrengthCategory PasswordStrengthCategory { get; set; }
+
+        public void CalculateMasterPasswordStrength(string password, string email)
+        {
+            var passwordStrength = _passwordGenerationService.PasswordStrength(password, GetPasswordStrengthUserInput(email));
+            if (string.IsNullOrEmpty(password))
+            {
+                PasswordStrength = 0;
+                PasswordColor = Utilities.ThemeManager.GetResourceColor("DangerColor");
+                PasswordStatus = " ";
+                PasswordStrengthCategory = PasswordStrengthCategory.Weak;
+                return;
+            }
```

Comment:
Yup i can move it up. It needs to have a space because on iOS if a field is empty it collapses the label, creating a weird UI flicker when you move between 0-1 characters.

Reply:
Roger, please add a comment explaining why the space is needed there.